e2e-tests/spring-rest-openapi-v2/src/test/java/org/evomaster/e2etests/spring/examples/displayissue/DisplayIssueEMTest.java

```diff
@@ -0,0 +1,144 @@
+    private String findValueOfItemWithKeyInStats(Solution s, String k) {
+
+        Statistics.Pair currentElement = null;
+        List<Statistics.Pair> stats= s.getStatistics();
+        Iterator<Statistics.Pair> statsIterator = stats.iterator();
+
+        while(statsIterator.hasNext()) {
+            currentElement = statsIterator.next();
+
+            if (currentElement.getHeader().equals(k)) {
+                return currentElement.getElement();
+            }
+        }
+
+        return null;
+
+    }
+
+    @Test
+    public void testRunBlackboxAndWhiteBox() throws Throwable {
+
+        // redirect console output to PrintStream
+
+        // old print stream
+        PrintStream old = System.out;
+
+        // run a white-box test using the controller first and check that
+        runTestHandlingFlakyAndCompilation(
+                "DisplayIssueEM",
+                "org.DisplayIssueEM",
+                1,
+                (args) -> {
+
+                    ByteArrayOutputStream byteArrayOutput = new ByteArrayOutputStream();
+                    System.setOut(new PrintStream(byteArrayOutput));
+
+                    //ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                    //PrintStream ps = new PrintStream(baos);
+
+                    // old print Stream
+                    //PrintStream old = System.out;
+
+
+                    // --blackBox true --bbSwaggerUrl https://petstore.swagger.io/v2/swagger.json
+                    // --outputFormat JAVA_JUNIT_4 --maxTime 60s --writeStatistics true
+                    args.add("--writeStatistics");
+                    args.add("true");
+
+
+                    Solution<RestIndividual> solutionWhite = initAndRun(args);
+
+                    // number of covered targets
+                    // the output should contain
+                    // * Covered targets (lines, branches, faults, etc.): {same value as the number of covered targets
+                    // from statistics} -
+                    // * Potential faults: {same value as the number of potential faults from statistics}
+                    String expectedNumberOfPotentialFaultsWhite = findValueOfItemWithKeyInStats(solutionWhite, "potentialFaults");
+                    String expectedNumberOfCoveredTargetsWhite = findValueOfItemWithKeyInStats(solutionWhite, "coveredTargets");
+
+                    Assert.assertTrue(byteArrayOutput.toString().contains("Covered targets " +
+                            "(lines, branches, faults, etc.): " + expectedNumberOfCoveredTargetsWhite));
+
+                    Assert.assertTrue(byteArrayOutput.toString().contains("Potential faults: "
+                            + expectedNumberOfPotentialFaultsWhite));
+
+                    // now run the same thins as a blackbox test and ensure that
+                    byteArrayOutput = new ByteArrayOutputStream();
+                    System.setOut(new PrintStream(byteArrayOutput));
+
+                    args.clear();
```

Comment:
this removes all the args set in `getArgsWithCompilation`

Reply:
Thank you so much Andrea for this comment. I have removed this line since I realized there is no need.